src/docfx/docset/Document.cs

```diff
@@ -241,40 +233,31 @@ public static Document TryCreateFromFile(Docset docset, string path)
 
         internal static ContentType GetContentType(string path)
         {
-            var name = Path.GetFileNameWithoutExtension(path);
-
-            if (path.EndsWith(".md", PathUtility.PathComparison))
+            if (!path.EndsWith(".md", PathUtility.PathComparison) &&
+                !path.EndsWith(".json", PathUtility.PathComparison) &&
+                !path.EndsWith(".yml", PathUtility.PathComparison))
             {
```

Comment:
exclude config file and extend file from here?

Reply:
that is a another workitem tracked in #3050